src/main/kotlin/racingcar/Application.kt

```diff
@@ -1,5 +1,97 @@
 package racingcar
 
+import camp.nextstep.edu.missionutils.Console
+import camp.nextstep.edu.missionutils.Randoms
+
 fun main() {
-    // TODO: 프로그램 구현
+    println("경주할 자동차 이름을 입력하세요.(이름은 쉼표(,) 기준으로 구분)")
+    val inputCar = Console.readLine()
+    var cars = inputCar.split(",")
+    isValidCarName(cars)
+    cars = duplicateName(cars)
+
+    println("시도할 횟수는 몇 회인가요?")
+    val inputMove = Console.readLine()
+    val move = isMovementValid(inputMove)
+    race(cars, move)
+}
+
+fun isValidCarName(cars: List<String>){
+    isOver5Letters(cars)
+    isNameless(cars)
+}
+
+fun isOver5Letters(cars: List<String>) {
+    cars.forEach{ car ->
+        if (5 < car.length) {
+            throw IllegalArgumentException("이름이 5자를 넘습니다")
+        }
+    }
+}
+
+fun isNameless(cars: List<String>) {
+    cars.forEach{ car ->
+        if (car.isBlank()) {
+            throw IllegalArgumentException("이름이 없는 참가자가 있습니다")
+        }
+    }
+}
+
+fun duplicateName(cars: List<String>): List<String> {
+    val carAndNumber = mutableMapOf<String, Int>()
+    val result = mutableListOf<String>()
+
+    cars.forEach{car ->
+        if (carAndNumber.containsKey(car)) {
+            val carNumber = carAndNumber[car]!! + 1
+            carAndNumber[car] = carNumber
+            result.add("$car ${carNumber}호기")
+        } else {
+            carAndNumber[car] = 1
+            result.add(car)
+        }
+    }
+    return result
+}
+
+fun isMovementValid(inputMove: String): UInt {
+    try {
+        val move = inputMove.toUInt()
+        return move
+    } catch (e: Exception) {
+        throw IllegalArgumentException("부적절한 이동 횟수 입니다")
+    }
```

Comment:
```suggestion
    val move = runCatching { inputMove.toUInt() }
        .getOrElse { throw IllegalArgumentException("부적절한 이동 횟수 입니다") }
    return move
```
코틀린에서 try-catch문을 대체하여 Result객체를 반환하는 runCatching{}도 이번에 알게 되었는데
공부해보시고 적용해보시면 좋을 것 같아서 이것도 남겨봅니다!

Reply:
Result 객체, runCatching{} 은 오늘 처음 알게 되었는데 확실히 간결해진 느낌이 듭니다!